src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpContext.cs

```diff
@@ -191,6 +197,38 @@ public string ReasonPhrase
             }
         }
 
+        public unsafe X509Certificate2 ClientCertificate
+        {
+            get
+            {
+                if (_options.ForwardClientCertificate &&
+                    _certificate == null &&
+                    NativeRequest->pSslInfo != null &&
+                    NativeRequest->pSslInfo->pClientCertInfo != null &&
+                    NativeRequest->pSslInfo->pClientCertInfo->pCertEncoded != null &&
+                    NativeRequest->pSslInfo->pClientCertInfo->CertEncodedSize != 0)
+                {
+                    try
```

Comment:
Just let it throw

Reply:
No, Getters shouldn't throw.

Reply:
Getters can totally throw. That's not against the rules. What sort of errors can this throw? It's kinda nasty catching all the exceptions.

We also have the async call to get the cert and that should throw, but the async call uses the sync API then it would also swallow the error no?

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property
"X AVOID throwing exceptions from property getters.

Property getters should be simple operations and should not have any preconditions. If a getter can throw an exception, it should probably be redesigned to be a method. Notice that this rule does not apply to indexers, where we do expect exceptions as a result of validating the arguments."

Reply:
Also, only HttpSys has a meaningful implementation of the async API, it can dynamically renegotiate the connection to request a cert on demand. Kestrel and IIS both require the cert up front.